src/catalog/src/durable/impls/persist.rs

```diff
@@ -1150,25 +1203,35 @@ impl DurableCatalogState for PersistCatalogState {
         boot_ts: mz_repr::Timestamp,
         _wait_for_consolidation: bool,
     ) -> Result<Vec<VersionedStorageUsage>, CatalogError> {
+        self.sync_to_current_upper().await?;
```

Comment:
Why do we need this if we didn't before?

Reply:
Previously we went straight to persist for a complete current snapshot, so we didn't need to update our in memory cache. Now, we serving from our cache so we need to make sure it's up to date.